FW/Presentations/Lab_Guides/_L_C_04_Writing_UEFI_App_Simics_Linux_Lab.md

```diff
@@ -0,0 +1,2024 @@
+
+```
+[PcdsFixedAtBuild]
+gEfiMdeModulePkgTokenSpaceGuid.PcdHelloWorldPrintTimes|3
+```
+
+Note: it is best to update PCD values in the Platform DSC file.
+
+2. Re-build BoardX58Ich10
+
+Open a termincal command prompt
+
+```bash
+$ cd ~/FW/edk2-ws/edk2-platforms/Platform/Intel/ 
+$ python build_bios.py -p BoardX58Ich10 -t GCC5
+```
+
+---
+## Slide 15 Update UefiAppLab.vhd File
+
+3. Mount the UefiAppLab.vhd using GuestMount: TENTATIVE LINK (See slide 87)
+
+4. Copy
+
+```bash
+$cp ~/FW/edk2-ws/Build/SimicsOpenBoardPkg/BoardX58Ich10/DEBUG_GCC5/X64/HelloWorld.efi ~/VHD
+```
+
+Paste (see powerpoint)
+
+
+---
+## Slide 16 EDK II HelloWorld App Solution
+
+5. Run Simics script
+
+```bash
+$> ./simics target/qsp-x86/qsp-modern-core.simics 
+simics> run
+```
+
+(Press "F2" at the logo, then select "Boot Manager" followed by "EFI Internal Shell")
+
+6. At the Shell prompt
+
+```
+Shell> Fs1:
+FS1:/> Helloworld
+UEFI Hello World!
+UEFI Hello World!
+UEFI Hello World!
+FS1:/>
+```
+
+7. Exit Simics
+
+```
+simics> Stop
+simics> quit
+```
+
+How can we change the **string** of the HelloWorld application?
+
+Also see `../edk2/Mde/ModulePkg.Dec`
+
+---
+## Slide 17 Lab 2: Write a Simple UEFI Application
+
+In this lab, you'll learn how to write simple UEFI applications
+
+---
+## Slide 18 LAB 2 Writing a Simple UEFI Application
+
+In this lab, you'll learn how to write simple UEFI applications.
+
+**"C" file**
+
+```
+EFI_STATUS
+EFIAPI
+UefiMain (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  return EFI_SUCCESS;
+}
+```
+
+**.inf file**
+
+```
+[Defines]
+  INF_VERSION       =
+  BASE_NAME         =
+  FILE_GUID 		=
+  MODULE_TYPE       =
+  VERSION_STRING    =             
+  ENTRY_POINT       =
+
+[Sources]
+  
+[Packages]
+    
+[LibraryClasses]
+```
+
+- What goes into a Simplest "C"
+- Start with what should go into the Simplest .INF file
+
+---
+## Slide 19 Application Lab - start with .c and .inf template
+
+Copy the `LabSampleCode/SampleApp/MyPkg` directory to `~/FW/edk2-ws/edk2`
+
+Edit **SampleApp.inf**
+
+- Look in the INF for "XXXXXXXXXXX" sections that will need information
+- Create Name & GUID, and then fill in the MODULE_TYPE
+
+---
+## Slide 20 Lab 2: Sample Application INF File
+
+```
+[Defines]
+  INF_VERSION         = 0x00010005
+  BASE_NAME           = XXXXXXXXXXXX
+  FILE_GUID           = XXXXXXXXXXXX
+  MODULE_TYPE         = XXXXXXXXXXXX
+  VERSION_STRING      = 1.0
+  ENTRY_POINT         = UefiMain
+
+[Sources]
+  XXXXXXXXX
+[Packages]
+  #XXXXXXXX
+  
+[LibraryClasses]
+  #XXXXXXXXXXXXX
+    
+[Guids]
+ # . . .
+```
+
+Get a GUID [guidgenerator.com/](https://www.guidgenerator.com/) or [https://www.guidgen.com/](https://www.guidgen.com/)
+
+Copy and paste [TENTATIVE LINK](https://github.com/tianocore-training/Presentation_FW/blob/main/FW/Presentations/Lab_Guides/_C_03_Writing_UEFI_App_WIN_LabGuide.md#slide-16-titlelab-2-sample-application-inf-file)
```

Comment:
This link is not needed since the PPTx / PDF will have a link to this spot in this md file.